plutus-core/plutus-core/src/PlutusCore/Evaluation/Machine/ExMemoryUsage.hs

```diff
@@ -168,19 +170,43 @@ instance ExMemoryUsage () where
     memoryUsage () = singletonRose 1
     {-# INLINE memoryUsage #-}
 
-{- | When invoking a built-in function, a value of type LiteralByteSize can be
-   used transparently as a built-in Integer but with a different size measure:
-   see Note [Integral types as Integer].  This is required by the
+{- | When invoking a built-in function, a value of type `NumBytesCostedAsNumWords`
+   can be used transparently as a built-in Integer but with a different size
+   measure: see Note [Integral types as Integer].  This is required by the
    `integerToByteString` builtin, which takes an argument `w` specifying the
    width (in bytes) of the output bytestring (zero-padded to the desired size).
    The memory consumed by the function is given by `w`, *not* the size of `w`.
-   The `LiteralByteSize` type wraps an Integer `w` in a newtype whose
+   The `NumBytesCostedAsNumWords` type wraps an Int `w` in a newtype whose
    `ExMemoryUsage` is equal to the number of eight-byte words required to
-   contain `w` bytes, allowing its costing function to work properly.
+   contain `w` bytes, allowing its costing function to work properly.  We also
+   use this for `replicateByte`.  If this is used to wrap an argument in the
+   denotation of a builtin then it *MUST* also be used to wrap the same argument
+   in the relevant budgeting benchmark.
 -}
-newtype LiteralByteSize = LiteralByteSize { unLiteralByteSize :: Integer }
-instance ExMemoryUsage LiteralByteSize where
-    memoryUsage (LiteralByteSize n) = singletonRose . fromIntegral $ ((n-1) `div` 8) + 1
+newtype NumBytesCostedAsNumWords = NumBytesCostedAsNumWords { unNumBytesCostedAsNumWords :: Int }
```

Comment:
Sounds like we should do bounds checking manually, since we're interested in a smaller range than `Integer` for bitwise builtins, so we don't really need to convert stuff to `Int` at the interface level, so this probably should be `Integer`. I don't know though and I don't think it should block this PR.

Reply:
I think I went back and forth on whether this should be `Integer` or `Int` and eventually came down on the side of `Int` because this currently gets used for `IntegerToByteString` and `replicateByte`, and in both of these it specifies a number of bytes (as the name suggests) and for that not to fit into an `Int` you'd be wanting more than something like 37 petabytes, which seems unlikely. If we made it into `Integer` then we'd have to add a range check and conversion to the implementation of `replicateByte`, but that's very fast anyway so the added cost wouldn't be important.  I'll merge this as is, but I may come back and change it to `Integer` for consistency.